lib/puppet/util/windows/user.rb

```diff
@@ -52,9 +52,20 @@ def check_token_membership
   module_function :check_token_membership
 
   def password_is?(name, password)
-    logon_user(name, password) { |token| }
-  rescue Puppet::Util::Windows::Error
-    false
+    # Windows API LogonUserW always disallows this with error 1327:
+    # Logon failure: user account restriction. Possible reasons are blank passwords
+    # not allowed, logon hour restrictions, or a policy restriction has been enforced.
+    if password.nil? || password == ''
+      raise Puppet::Util::Windows::Error.new(
+        "Windows does not have the ability to verify NULL or empty passwords"
+      )
+    end
```

Comment:
Oh right, now I remember why.  New users call into the ADSI create directly, and the provider code path for password isn't touched.

But when you modify an existing user:

```
bundle exec puppet apply -e "user { 'ramon': managehome => false, password => '' }"
DL is deprecated, please use Fiddle
Notice: Compiled catalog for vagrant-2008r2.corp.puppetlabs.net in environment production in 0.06 seconds
Warning: Blank (zero length) passwords are not recommended. Blank passwords may be allowed on your system based on policy, but Puppet cannot verify a blank password with the operating system. Because of this a password change event will be executed by Puppet on every run.
Notice: /Stage[main]/Main/User[ramon]/password: changed password
Notice: Applied catalog in 0.22 seconds
```

I think we need to move the location of the warning.